lib/elixir/lib/calendar/datetime.ex

```diff
@@ -1674,6 +1675,128 @@ defmodule DateTime do
     end
   end
 
+  @doc """
+  Shifts given `datetime` by `duration` according to its calendar.
+
+  Allowed units are: `:year`, `:month`, `:week`, `:day`, `:hour`, `:minute`, `:second`, `:microsecond`.
+
+  Shifting datetimes in time zones that observe "Daylight Saving Time" across
+  summer/winter time will add/remove one hour from the resulting datetime.
+  This ensures `shift/3` always returns a valid datetime.
+
+  Durations are collapsed before they are applied:
+  - when shifting by 1 year and 2 months the date is actually shifted by 14 months
+  - weeks, days and smaller units are collapsed into seconds and microseconds
+
+  Durations are applied in order of the size of the unit: `month > second > microsecond`.
+
+  ## Examples
+
+      iex> DateTime.shift(~U[2016-01-01 00:00:00Z], month: 2)
+      ~U[2016-03-01 00:00:00Z]
+      iex> DateTime.shift(~U[2016-01-01 00:00:00Z], year: 1, week: 4)
+      ~U[2017-01-29 00:00:00Z]
+      iex> DateTime.shift(~U[2016-01-01 00:00:00Z], minute: 25)
+      ~U[2016-01-01 00:25:00Z]
+      iex> DateTime.shift(~U[2016-01-01 00:00:00Z], minute: 5, microsecond: {500, 4})
+      ~U[2016-01-01 00:05:00.0005Z]
```

Comment:
I tried these out of curiosity, maybe it would help to have the examples in the doc?
```suggestion
      # leap years
      iex> DateTime.shift(~U[2024-02-29 00:00:00Z], year: 1)
      ~U[2025-02-28 00:17:35Z]
      iex> DateTime.shift(~U[2024-02-29 00:00:00Z], year: 4)
      ~U[2028-02-29 00:17:35Z]
```

Reply:
done in https://github.com/elixir-lang/elixir/commit/5e9d6fbe5f25e3c5a263f825aa7266c5190f2742 and https://github.com/elixir-lang/elixir/pull/13385/commits/0f8957ee45e58245ba20bd8f8aef382a3c0e93ce